wgpu-core/src/device/queue.rs

```diff
@@ -42,14 +42,59 @@ use super::Device;
 pub struct Queue {
     raw: ManuallyDrop<Box<dyn hal::DynQueue>>,
     pub(crate) device: Arc<Device>,
+    pub(crate) pending_writes: Mutex<ManuallyDrop<PendingWrites>>,
```

Comment:
question: RE: the `move PendingWrites into the Queue` commit: The commit message claims that it avoids (removes?) strong ref. cycles between a `Device` and resources containing backlinks to it. I don't see how moving `pending_writes` from a `Device` to a `Queue` (which seems to be held by a `Device`, and also hold a strong ref. to a `Device) does anything other than increase the number of steps in a cycle. Am I missing something?

Reply:
Ah, I think I understand the answer: a `Queue` is held behind a `Weak` ref., rather than an `Arc`, in a `Device`. So, this should break a cycle, because the ref. link to `Queue` is comparatively weaker.